### PR TITLE
Finally fix containsOne/Many UI property overwrites

### PR DESCRIPTION
**Pull request by DarkSide666**

### CODE REVIEW

**Review by ibelar — APPROVED** [review]
LGTM

File: src/Reference/ContainsOne.php [hunk 1]
```diff
@@ -60,14 +57,18 @@ protected function init(): void
...
-        if (!$ourModel->hasElement($this->our_field)) {
-            $ourModel->addField($this->our_field, [
+        if (!$ourModel->hasElement($ourField)) {
+            $ourModel->addField($ourField, [
                 'type' => $this->type,
                 'reference' => $this,
                 'system' => $this->system,
                 'caption' => $this->caption, // it's ref models caption, but we can use it here for field too
-                'ui' => $this->ui,
+                'ui' => array_merge_recursive([
```
Comment (mvorisek) [inline src/Reference/ContainsOne.php:68]: array_merge_recursive DOES NOT have any affect here -> use `array_merge`

Comment (DarkSide666) [reply src/Reference/ContainsOne.php:68]: `ui` is multi-level array property. array_merge will merge only 1st level of array and overwrite all other levels, so it's not possible to extend `ui` with `ui => ['form' => [AnotherClass::class]]` for example without breaking other (maybe default ones) ui properties.

OH crap, you're right. It should be `array_replace_recursive` here - I will fix.

```
$a = ['form' => ['MyForm', 'color'=>'red'], 'table' => ['color'=>'blue']];
$b = ['form' => ['color'=>'green'], 'table'=>['MyTable']];
print_r($c = array_merge_recursive($a,$b)); // will create color as array [red, green] :(
print_r($d = array_replace_recursive($a,$b)); // will correctly replace color
assert($d, ['form' => ['MyForm', 'color'=>'green'], 'table' => ['MyTable', 'color'=>'blue']]);
```
fixed in https://github.com/atk4/data/pull/789

Comment (mvorisek) [reply src/Reference/ContainsOne.php:68]: I did not read, but here the 1st arg is NOT nested arrray, thus I do not understand `_recursive`, seems like a typo to me

**Pull request merged**